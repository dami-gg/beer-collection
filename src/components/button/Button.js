import React from 'react';

import './button.scss';

const Button = (props) => (
    <button type={props.type || "button"}
            className={`button button--${props.color} ${props.classes}`}
            onClick={event => props.onClick && props.onClick(event)}>
      {props.children}
    </button>
);

export default Button;
