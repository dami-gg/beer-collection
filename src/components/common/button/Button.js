// @flow

import React, {PureComponent} from 'react';

import './button.scss';

class Button extends PureComponent {
  props: {
    type?: string,
    color: string,
    classes?: string,
    onClick: Function,
    children: any
  };

  render() {
    return (
        <button type={this.props.type || "button"}
                className={`button button--${this.props.color} ${this.props.classes}`}
                onClick={event => this.props.onClick && this.props.onClick(event)}>
          {this.props.children}
        </button>
    );
  }
}

export default Button;
