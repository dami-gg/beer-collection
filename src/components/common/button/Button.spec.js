import React from 'react';
import Button from './Button';

import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';

describe('Button', () => {
  let component,
      element,
      mock;

  beforeEach(mocks);

  it('should render without crashing', () => {
    shallow(<Button></Button>);
  });

  it('should match the snapshot', () => {
    component = renderer.create(<Button></Button>);
    let json = component.toJSON();

    expect(json).toMatchSnapshot();
  });

  describe('render', () => {
    beforeEach(setup);

    it('should render a button of the type specified in the props', () => {
      element = component.find('button');

      expect(element.props().type).toBe(mock.props.type);
    });

    it('should render a button of the type specified in the props', () => {
      element = component.find('button');
      element.simulate('click');

      expect(mock.props.onClick).toHaveBeenCalled();
    });

    it('should render a button with the class specified in props', () => {
      element = component.find('button');

      expect(element.hasClass(mock.props.classes)).toBe(true);
    });

    it('should render a button of the color specified in props', () => {
      element = component.find('button');

      expect(element.hasClass(`button--${mock.props.color}`)).toBe(true);
    });
  });

  function setup() {
    render();
  }

  function render() {
    component = shallow(
        <Button
            type={mock.props.type}
            color={mock.props.color}
            classes={mock.props.classes}
            onClick={mock.props.onClick}>
        </Button>
    );
  }

  function mocks() {
    mock = {
      props: {
        type: 'type',
        color: 'color',
        classes: 'classes',
        onClick: jest.fn()
      }
    };
  }
});

