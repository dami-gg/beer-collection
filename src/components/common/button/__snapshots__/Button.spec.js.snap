exports[`Button should match the snapshot 1`] = `
<button
  className="button button--undefined undefined"
  onClick={[Function onClick]}
  type="button" />
`;
